src/Signer.php

```diff
@@ -35,6 +35,16 @@ public function sign($message)
     {
         $signatureParameters = $this->signatureParameters($message);
         $message = $message->withAddedHeader("Signature", $signatureParameters->string());
+        return $message;
+    }
+
+    /**
+     * @param RequestInterface $message
+     * @return RequestInterface
+     */
+    public function authorize($message)
+    {
+        $signatureParameters = $this->signatureParameters($message);
         $message = $message->withAddedHeader("Authorization", "Signature " . $signatureParameters->string());
         return $message;
     }
```

Comment:
They have the same cryptographic signatures if signing the same data (headers) but serve two different intents, and the distinction is crucial:

* `Authorization` is a claim that I am authorised to make a statement. For a token request, it's "I am a particular identity, I can prove it with this signature derived from a secret only I know, and would like to perform an action.". It could be a signature, a username/password, APIKey etc, but is generally proof of identity through proof of knowledge/possession. The signature is incidental to the purpose.
* `Signature` is an assertion that information is correct - including the request I am making `(request-target)`, headers, signed body etc) has not changed during transmission - something not explicitly asserted in an authorisation claim. The requester typically accompanies this with a separate `Authorization` artefact (e.g. a Bearer token, or indeed maybe a HTTP signature but it is distinct and may be signed differently to convey the different intent).
* The `Signature` is especially important for some requests e.g. My Signature asserts I have correctly formatted the body of my POST action to initiate a payment to someone - you can trust I have actually sent this and crucially I cannot deny I sent it at a later time (non-repudiation).

Crucially a specific request may *require* an `Authorization` header with e.g. a Bearer token, which we would either overwrite or add a second instance of using this library. See for example [page 36 of this standard](https://docs.wixstatic.com/ugd/c2914b_5351b289bf844c6881e46ee3561d95bb.pdf) - the `Authorization` header carries an unrelated token (to authorise the request) alongside a `Signature` header (which validates the content of the request and that it was explicitly and intentionally made by the requester with the force of a contract).

Even outside implementation constraints lumping them together doesn't force the requester to think about exactly which claim they are making. "It just works" serves nobody in Information Security, and it's unlikely an endpoint would require both headers under the HTTP Signatures scheme. If so, explicitly request both with `->sign()` and `->authorize()`, for each intent (specifically, Contexts).

Apologies for the long explanation, I'm actually an InfoSec professional not a developer, and getting this right matters to me.

Reply:
> What actually is the difference between `Signature` and `Authorization` anyway? Seems like the same data is on each header.

@mtibben Please check https://github.com/w3c-dvcg/http-signatures/pull/48 and see if it is clear, these are quite distinct so important to make sure it's right and communicated properly for non-specialists.

Reply:
I'm consulting on an implementation now where the Authorization header uses a (bearer) token retrieved from an OAuth2 service, and Signature header from this protocol. Using this library as-is (generating both) would either overwrite the Authorization header (losing the token) or add a second Authorization header, which most implementations would probably not understand or some may outright reject if they're not expecting it.

This is a required change IMHO, though I realise the potential for BC impact.

Reply:
This distinction is now part of the RFC version 11, clearly spelled out in [section 1](https://tools.ietf.org/html/draft-cavage-http-signatures-11#section-1). Since they serve distinct purposes, and can collide with existing headers, I don't see how we can carry on generating both in all cases.